tests/nn_test.py

```diff
@@ -61,6 +61,33 @@ def testHardTanhMemory(self):
     # see https://github.com/google/jax/pull/1640
     jax.make_jaxpr(nn.hard_tanh)(np.ones((10 ** 12,)))  # don't oom
 
+  def testOneHot(self):
+    actual = nn.one_hot(np.array([0, 1, 2]), 3)
```

Comment:
Maybe add a non-regular example, e.g., `one_hot([2, 0], 3)` ?

Reply:
Done, also fixed the f64 test case.